src/main/java/net/mcreator/generator/blockly/BlocklyBlockCodeGenerator.java

```diff
@@ -118,6 +119,8 @@ public void generateBlock(BlocklyToCode master, Element block) throws TemplateGe
 						found = true;
 						String generatedCode = BlocklyToCode.directProcessOutputBlock(master, element);
 						dataModel.put("input$" + inputName, generatedCode);
+						// We also pass the type of the block that is attached to the input

```

Comment:
I am not sure if calling this "type" is the best, as we use type already for:

* determining type of the block (ToolboxBlock#type) - procedural or output
* especially output blocks, that are used in inputs have return _types_ such as logic, number, ...

What you call "type" here is more properly called ToolboxBlock#machine_name so I would suggest renaming type in template data model and comments here to something more fitting to reduce the similar name reuse for different purposes

Reply:
Fair, I originally went with "type" as it's called that way in JS and xml. I guess "id" could be a valid alternative that isn't too long